### ISSUE TITLE
Sharp package built with wrong NODE_MODULE_VERSION when deploying using meteor-up

### ISSUE THREAD

**Issue by Neijin** [zodern/meteor-up#1005]
Hi guys ! Thank you for all the good work you are doing here.

## **Context**
I'm trying to deploy my meteor project on production using [zodern/meteor-up].(https://github.com/zodern/meteor-up).

**_Dev_** : 
Windows 10 (x64)
meteor 1.7.0.5 
node v8.11.4 (only one version installed)
**Mup version (`mup --version`):** 1.4.5
**Mup config**

```js
{
  "servers": {
    "one": {
      "host": "1.2.3.4",
      "username": "root",
      "pem": "~/.ssh/pem"
    }
  },
  "app": {
    "name": "my-app",
    "path": "<app-path>",
    "type": "meteor",
    "volumes": {
    },
    "docker": {
      "image": "abernix/meteord:node-8.11.2-base",
      "prepareBundle": true,
      "buildInstructions": [
      ],
      "stopAppDuringPrepareBundle": true
    },
    "servers": {
      "one": {
        "env": {}
      }
    },
    "buildOptions": {
      "serverOnly": true,
      "debug": false
    },
    "env": {
      "ROOT_URL": "https://subdomain.host.com",
      "MONGO_URL": "mongodb://user:pass@subdomain.subdomain.host.com:27017/production",
      "NODE_ENV": "production",
      "VIRTUAL_HOST": "subdomain.host.com",
      "HTTPS_METHOD": "redirect",
      "HTTP_FORWARDED_COUNT": 1,
      "LETSENCRYPT_HOST": "subdomain.host.com",
      "LETSENCRYPT_EMAIL": "host.com@gmail.com"
    },
    "enableUploadProgressBar": true
  },
  "proxy": {
    "domains": "subdomain.host.com",
    "ssl": {
      "letsEncryptEmail": "host.com@gmail.com",
      "forceSSL": true
    }
  }
}
```

**_Prod_** : 
Docker instance (abernix/meteord:node-8.11.2-base image) running on Debian 9
node v8.11.2

Output of command

```bash
paste output here...
```

## **Error**
Everything looks successfull until meteor starts. Then this error occurs:

```js
/built_app/programs/server/node_modules/fibers/future.js:280
                                                throw(ex);
                                                ^

Error: The module '/built_app/programs/server/npm/node_modules/sharp/build/Release/sharp.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 46. This version of Node.js requires
NODE_MODULE_VERSION 57. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or `npm install`).
    at Object.Module._extensions..node (module.js:681:18)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/built_app/programs/server/npm/node_modules/sharp/lib/constructor.js:10:15)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/built_app/programs/server/npm/node_modules/sharp/lib/index.js:3:15)
    at Module._compile (module.js:652:30)
```

I tried a lot a different approaches during these past two days without success.
Do you guys have any ideas ?

**Comment by SimonSimCity** [comment]
Looks like you somewhere have NodeJS in some `4.x` version installed - otherwise I wonder how it could be built with this version. My advise:

1. Remove `node_modules` folder of your project
1. Run `meteor npm install --production`

And again check your node versions within meteor and on the system installed, which seems the one mup uses.

**Comment by Neijin** [comment]
Hi @SimonSimCity. Thank you for your response.

**I got it working by pushing from an MacOSX dev environment.** Sharp is then compiled with the right NODE_VERSION on the production server. I'm clueless on how to resolve this issue on Windows and why there is a difference. 

I checked for any forgotten installation of Node 4 pretty much anywhere I could think of (this is [a great reminder by the way on how to completely remove node.js from Windows](https://stackoverflow.com/questions/20711240/how-to-completely-remove-node-js-from-windows)).

I already tried to remove the node_modules folder multiple times and re-install npm modules with different options. There was also an issue with the node_modules/sharp/vendor folder keeping old compiled files from previous build so it was recommended to delete it before pushing to production.
